kousa/lib/kousa/room.ex

```diff
@@ -320,7 +358,9 @@ defmodule Kousa.Room do
                      room.id,
                      {:redeem_invite, user_id}
                    ) do
-                {:ok, :error} -> {:err, "the room is private, ask someone inside to invite you"}
+                {:ok, :error} -> 
+                      send_waiting_room_casts(user_id, room)
+                      exit(:shutdown)
```

Comment:
why did you do this?

Reply:
To exit the function. Otherwise the user will automatically join the room